### PR TITLE
Introduce role to application rp configuration.

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* Role doesn't support hot load.
* Role can be overriden by environment variables.

Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

**Review by trask — APPROVED** [review]
thx @Incarnation-p-lee!

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java [hunk 1]
```diff
@@ -285,6 +285,14 @@ private static void overlayRpConfiguration(Configuration config, RpConfiguration
         if (rpConfiguration.ignoreRemoteParentNotSampled != null) {
             config.preview.ignoreRemoteParentNotSampled = rpConfiguration.ignoreRemoteParentNotSampled;
         }
+        if (isTrimEmpty(config.role.name)) {
+            // only use rp configuration role name as a fallback, similar to WEBSITE_SITE_NAME
+            config.role.name = rpConfiguration.role.name;
+        }
+        if (isTrimEmpty(config.role.instance)) {
+            // only use rp configuration role name as a fallback, similar to WEBSITE_INSTANCE_ID
+            config.role.instance = rpConfiguration.role.instance;
```
Comment (Incarnation-p-lee) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java:294]: If so, looks customer cannot override this by env variables ?

Comment (Incarnation-p-lee) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java:294]: Azure Spring Cloud will set the role name from applicationinsights-rp.json by default. But we should allow customer to override the default behavior by env variables like ROLE_NAME and ROLE_INSTANCE.

Currently looks rpConfiguration will always win, right ?

**Comment by heyams** [comment]
We need to update azure docs, right? @trask.  I am going to merge it in.

**Comment by Incarnation-p-lee** [comment]
@heyams  Please wait a moment ? Looks there is one question as above need to address with @trask .

**Comment by trask** [comment]
it will only set role name/instance from rp if it wasn't set by user, I sent PR to add test for this #1663, have a look and let us know if you still see any issue, thx!

**Comment by Incarnation-p-lee** [comment]
Great! @trask just look back your changes, it is what we wanted. Thanks a lot.

**Pull request merged**